staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiservice_controller.go

```diff
@@ -80,12 +80,6 @@ func (c *APIServiceRegistrationController) sync(key string) error {
 		return err
 	}
 
-	// remove registration handling for APIServices which are not available
```

Comment:
We have to return 502 or 503 instead. Is that the case? We need a test for that.

I remember we had aggregator issues like long-running not terminating requests when the apiserver was unavailable. If I remember right, this code block was added to remedy that. But I agree that a 404 is not a good response at all. @deads2k

Reply:
Yes, 503 is a better response. If a 404, it means the object was deleted from etcd. My controller may begin to delete all other resources related to this object. It is very dangerous

Reply:
Maybe the garbage collector will do the same on 404. Not good.